### PR TITLE
[table64] Fix generate-dyncalls and directize under table64

### PR DESCRIPTION
**Pull request by sbc100**

### CODE REVIEW

File: test/lit/passes/generate-dyncalls-wasm64.wast [hunk 1]
```diff
@@ -0,0 +1,52 @@
+;; NOTE: Assertions have been generated by update_lit_checks.py --all-items and should not be edited.
+;; RUN: wasm-opt %s --generate-dyncalls --enable-memory64 --table64-lowering -S -o - | filecheck %s
```
Comment (kripken) [inline test/lit/passes/generate-dyncalls-wasm64.wast:2]: Why add lowering here?

**Review by kripken — APPROVED** [review]
These could be separate PRs as they are independent? But they are also trivial so sgtm.

The title could maybe be `[Table64] Fix generate-dyncalls and directize` or such?

File: test/lit/passes/generate-dyncalls-wasm64.wast (hunk 1 above)
Comment (sbc100) [reply test/lit/passes/generate-dyncalls-wasm64.wast:2]: Hmm.. I think you are right.. I need to make sure that it still fails without that patch in that case.

Comment (sbc100) [reply test/lit/passes/generate-dyncalls-wasm64.wast:2]: Yup, you are correct.  The orignal crash and the fix both confirmed without the need for lowering here.

**Review by kripken — APPROVED** [review]

**Pull request merged**